### PR TITLE
Add tests for all exceptions

### PR DESCRIPTION
**Pull request by joeirimpan**
@vividvilla Please review and merge

### CODE REVIEW

File: kiteconnect/connect.py [hunk 1]
```diff
@@ -645,11 +645,8 @@ def _request(self, route, method, parameters=None):
                         return
 
                 # native Kite errors
-                exp = getattr(ex, data["error_type"])
-                if exp:
-                    raise(exp(data["message"], code=r.status_code))
-                else:
-                    raise(ex.GeneralException(data["message"], code=r.status_code))
+                exp = getattr(ex, data["error_type"], ex.GeneralException)
```
Comment (vividvilla) [inline kiteconnect/connect.py]: Good one :clap:

File: kiteconnect/connect.py [hunk 2]
```diff
@@ -634,8 +634,8 @@ def _request(self, route, method, parameters=None):
             try:
                 data = json.loads(r.content.decode("utf8"))
             except:
-                raise ex.DataException("Couldn't parse the JSON response received from the server: {contnet}".format(
-                    content=data))
+                raise ex.DataException("Couldn't parse the JSON response received from the server: {content}".format(
+                    content=r.content))
```
Comment (vividvilla) [inline kiteconnect/connect.py]: Since this is a bug It would have been better if its a separate commit.

File: kiteconnect/connect.py (hunk 1 above)
Comment (vividvilla) [inline kiteconnect/connect.py]: Good one :clap:

File: kiteconnect/connect.py (hunk 2 above)
Comment (joeirimpan) [reply kiteconnect/connect.py]: @vividvilla Did you mean separate PR?

Comment (vividvilla) [reply kiteconnect/connect.py]: This is fine, I didn't notice that these were different commits since I was checking files changed. All good :thumbsup:

**Pull request merged**